pkg/chains/storage/tekton/tekton.go

```diff
@@ -110,18 +111,37 @@ func (b *Backend) retrieveAnnotationValue(annotationKey string, decode bool) (st
 }
 
 // RetrieveSignature retrieve the signature stored in the taskrun.
-func (b *Backend) RetrieveSignature(opts config.StorageOpts) (string, error) {
+func (b *Backend) RetrieveSignatures(opts config.StorageOpts) (map[string][]string, error) {
 	b.logger.Infof("Retrieving signature on TaskRun %s/%s", b.tr.Namespace, b.tr.Name)
-	return b.retrieveAnnotationValue(fmt.Sprintf(SignatureAnnotationFormat, opts.Key), true)
+	signature, err := b.retrieveAnnotationValue(fmt.Sprintf(SignatureAnnotationFormat, opts.Key), true)
+	if err != nil {
+		return nil, err
+	}
+
+	m := make(map[string][]string)
+	for _, res := range b.tr.Status.TaskRunResults {
+		if strings.HasSuffix(res.Name, "IMAGE_URL") {
+			m[res.Value] = []string{signature}
```

Comment:
should we just be mapping [SignatureAnnotation]:signature for the `tekton` case?

Reply:
Sure thing! Addressed ✅ 

Should we also add the check to ensure there is only one image then?

Reply:
i think this should be ok :)